tests/integration/test_index_routines.py

```diff
@@ -102,6 +107,160 @@ def test_choose_target_ndim(ndim):
                 assert np.array_equal(np_res, num_res)
 
 
+SHAPES_A = (
+    (2, 4),
+    (2, 1),
+    (1, 4),
+    (1, 1),
+    (4,),
+    (1,),
+    (3, 2, 4),
+    (2, 3, 2, 4),
+    (1, 3, 1, 1),
+)
+
+
+@pytest.mark.parametrize(
+    "shape_a", SHAPES_A, ids=lambda shape_a: f"(shape_a={shape_a})"
+)
+def test_choose_a_array(shape_a):
+    shape_choices = (3, 2, 4)
+    np_a = mk_seq_array(np, shape_a) % shape_choices[0]
+    num_a = mk_seq_array(num, shape_a) % shape_choices[0]
+    np_choices = mk_seq_array(np, shape_choices)
+    num_choices = mk_seq_array(num, shape_choices)
+
+    np_res = np.choose(np_a, np_choices)
+    num_res = num.choose(num_a, num_choices)
+    assert np.array_equal(np_res, num_res)
+
+
+def test_choose_a_scalar():
+    shape_choices = (3, 2, 4)
+    a = 1
+    np_choices = mk_seq_array(np, shape_choices)
+    num_choices = mk_seq_array(num, shape_choices)
+
+    np_res = np.choose(a, np_choices)
+    num_res = num.choose(a, num_choices)
+    assert np.array_equal(np_res, num_res)
+
+
+@pytest.mark.parametrize("mode", ("wrap", "clip"), ids=str)
+@pytest.mark.parametrize(
+    "shape_a", ((3, 2, 4), (4,)), ids=lambda shape_a: f"(shape_a={shape_a})"
+)
+def test_choose_mode(shape_a, mode):
+    shape_choices = (3, 2, 4)
+    np_a = mk_seq_array(np, shape_a) - 10
+    num_a = mk_seq_array(num, shape_a) - 10
+    np_choices = mk_seq_array(np, shape_choices)
+    num_choices = mk_seq_array(num, shape_choices)
+
+    np_res = np.choose(np_a, np_choices, mode=mode)
+    num_res = num.choose(num_a, num_choices, mode=mode)
+    assert np.array_equal(np_res, num_res)
+
+
+def test_choose_out():
+    shape_choices = (3, 2, 4)
+    shape_a = (2, 4)
+    shape_a_out = (2, 4)
+    np_a = mk_seq_array(np, shape_a) % shape_choices[0]
+    np_a = np_a.astype(np.int32)
+    num_a = mk_seq_array(num, shape_a) % shape_choices[0]
+    num_a = num_a.astype(
+        np.int32
+    )  # cuNumeric would convert np.int32 to default type np.int64
+    np_choices = mk_seq_array(np, shape_choices)
+    num_choices = mk_seq_array(num, shape_choices)
+    np_aout = mk_seq_array(np, shape_a_out) - 10
+    num_aout = mk_seq_array(num, shape_a_out) - 10
+
+    np_res = np.choose(np_a, np_choices, out=np_aout)
+    num_res = num.choose(num_a, num_choices, out=num_aout)
+    assert np.array_equal(np_res, num_res)
+    assert np.array_equal(np_aout, num_aout)
+
+
+@pytest.mark.xfail
+def test_choose_mode_none():
+    # In Numpy, pass and returns array equals default mode
+    # In cuNumeric, raises ValueError: mode=None not understood.
```

Comment:
Please open an issue about this, it seems like it should be an easy fix to handle None